### DISCUSSION TITLE
Question about __init__ as __init__

### DISCUSSION THREAD

**Discussion by PTank** [fastapi/fastapi#4808] [Questions] [answered]
## Why import X as  X

I've check the fastapi's code, and i saw the __init__.py file 

https://github.com/tiangolo/fastapi/blob/master/fastapi/__init__.py
```python
"""FastAPI framework, high performance, easy to learn, fast to code, ready for production"""

__version__ = "0.75.2"

from starlette import status as status

from .applications import FastAPI as FastAPI
from .background import BackgroundTasks as BackgroundTasks
from .datastructures import UploadFile as UploadFile
from .exceptions import HTTPException as HTTPException
from .param_functions import Body as Body
from .param_functions import Cookie as Cookie
from .param_functions import Depends as Depends
from .param_functions import File as File
from .param_functions import Form as Form
from .param_functions import Header as Header
from .param_functions import Path as Path
from .param_functions import Query as Query
from .param_functions import Security as Security
from .requests import Request as Request
from .responses import Response as Response
from .routing import APIRouter as APIRouter
from .websockets import WebSocket as WebSocket
from .websockets import WebSocketDisconnect as WebSocketDisconnect
```

My ask is why all import use the key word "as" to be named with the same name ?

**Comment by vrslev** [comment, marked as answer]
`from … import X as X`  is equivalent of adding `X` to `__all__` variable. This is required for type checking and autocompletion.

**Reply by PTank** [reply]
I miss something I guess, I saw no difference with `from .. import X` and `from .. import X as X` even if I check for an `import *` or autocompletion. If you have a link about that I'm curious to know when I have to do that.

Thanks !

**Reply by vrslev** [reply]
It starts to make a difference when you build a library, not just an application where all the code is in one place.
[pyright](https://github.com/microsoft/pyright/blob/fd98a8a2b964056f7afc616896735689c7cba421/docs/typed-libraries.md#library-interface), [mypy](https://mypy.readthedocs.io/en/stable/config_file.html?highlight=import%20as#confval-implicit_reexport)
